lib/foreman/renderer.rb

```diff
@@ -26,6 +26,18 @@ def template_logger
       @template_logger ||= Foreman::Logging.logger('templates')
     end
 
+    def host_info(*path)
+      @info ||= @host.info
+      info = @info
+      path.each { |step| info = info.fetch step }
+      info
+    end
+
+    def host_param(*path)
```

Comment:
I think we agreed some time ago that we want to avoid using instance variables directly such as `@host`. IMHO calling their methods is also problematic. Things like `@host.info` or `@host.operatingsystem` are internal Foreman methods, if we encourage people to use them in templates, we can't modify them without templates modifications. Imagine we would like to rename `Host#operatingsystem` to `Host#operating_system`. Yes, macros can be seen as a duplication but they isolate template writers from internal changes and help to define templates API.

There are other benefits of macros over internal methods, e.g. it's easier to extend macros from plugins than modifying `@host` variable. Also we could better support previewing since the method can contain logic to handle nil objects so if `@host` is nil we could render just string '$host' or something like that.